### PR TITLE
JUnit task done

### PR DESCRIPTION
**Pull request by less08**

### CODE REVIEW

File: src/main/java/core/basesyntax/Calculator.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
+
+public class Calculator {
+
+    public double calculate(double a, double b, char operator) {
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/Calculator.java:5]: let's name the variables more informative. like `firstNumber`

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class CalculatorTest {
+    Calculator calculator;
+
+    @BeforeEach
+    private void beforeEach() {
+        calculator = new Calculator();
+    }
```
Comment (Max-Pochepets) [inline src/test/java/core/basesyntax/CalculatorTest.java:15]: do u really need to create a new object `@BeforeEach` test?

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        assertEquals(-8, actual);
+    }
+
+    @Test
+    void raiseNegativeValueToNegativePow() {
+        double actual = calculator.calculate(-2, -3, '^');
+        assertEquals(-0.125, actual);
+    }
+
+    //- raising positive/negative value to zero power
```
Comment (Max-Pochepets) [inline src/test/java/core/basesyntax/CalculatorTest.java:160]: delete this comment, the method name should speak for itself :)

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 4]
```diff
@@ -0,0 +1,183 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class CalculatorTest {
```
Comment (Max-Pochepets) [inline src/test/java/core/basesyntax/CalculatorTest.java:9]: how the program should behave if we, for example, add 1 to `Integer.MAX_VALUE` (aka edge cases) etc.

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/CalculatorTest.java (hunk 4 above)
Comment (Artificial-Friend) [reply src/test/java/core/basesyntax/CalculatorTest.java:9]: Double.MAX_VALUE is more appropriate since method receives double and can fit Integer.
Therefore I recommend to test INFINITY since method can accept it.

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 5]
```diff
@@ -0,0 +1,181 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class CalculatorTest {
+    Calculator calculator = new Calculator();
```
Comment (Max-Pochepets) [inline src/test/java/core/basesyntax/CalculatorTest.java:9]: 0_0 does it work? I meant you didn't need `BeforeEach` in terms of performance, but u need smth else.

Comment (Max-Pochepets) [reply src/test/java/core/basesyntax/CalculatorTest.java:9]: if this works, then nvm

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    @Test
+    void illegalOperation() {
+        assertThrows(RuntimeException.class, () -> calculator.calculate(10, 0, '?'));
+    }
+
+    @Test
+    void doubleOverflow() {
+        assertThrows(RuntimeException.class, () -> calculator.calculate(Double.MAX_VALUE, 1, '+'));
```
Comment (Max-Pochepets) [inline src/test/java/core/basesyntax/CalculatorTest.java:179]: add edge cases for other operations, like what if we `Double.MIN_VALUE - 1`?

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

**Review by Max-Pochepets — APPROVED** [review]

**Comment by less08** [comment]
Changed the argument type for all methods from double to integer. Now the only possible way to overflow double is Multiply Integer.Max_VALUE * Integer. This case is covered now

File: src/main/java/core/basesyntax/Calculator.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        BigDecimal bigDecimal = new BigDecimal(firstValue).multiply(new BigDecimal(secondValue));
+        if (bigDecimal.compareTo(new BigDecimal((double) firstValue * secondValue)) == 0) {
+            return bigDecimal.doubleValue();
+        }
+        throw new ArithmeticException("Double overflow");
+    }
+
+    private double divide(int firstValue, int secondValue) {
+        if (secondValue == 0) {
+            throw new ArithmeticException("Division  secondValue by 0");
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Calculator.java:42]:
```suggestion
            throw new ArithmeticException("Division secondValue by 0");
```

File: src/main/java/core/basesyntax/Calculator.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        return (double) firstValue / secondValue;
+    }
+
+    private double pow(int firstValue, int secondValue) {
+        return Math.pow(firstValue, secondValue);
+    }
+
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Calculator.java:50]: redundant empty line

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 9]
```diff
@@ -0,0 +1,182 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class CalculatorTest {
+    Calculator calculator = new Calculator();
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:9]: let’s create a Calculator instance in a method that is run before all tests

add access modifiers to all class components

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
for each operation, let's also test passing Integer.MAX/MIN_VALUE

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
also, please move your solution to a separate branch and create a new PR - currently you are working in master
